src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -102,6 +102,21 @@ def exit_code(self):
     return self._exit_code
 
 
+class _PantsProductPrecomputeFailed(Exception):
+  """
+  Represents a failure in the precompute step of v2 products for pants runs.
+  (e.g. The user called ./pants dependees and there was a broken link)
+
+  Will be raised by the DaemonPantsRunner at creation time.
+  """
+
+  def __init__(self, wrapped_exception):
+    """
+    :param wrapped_exception: Original exception raised by the engine.
+    """
+    super().__init__('Product precomputation in the daemon raised an exception: {}'.format(wrapped_exception))
```

Comment:
F strings :)

Reply:
Also, I think from Pycon they said to always use the `repr` for exceptions to ensure that the exception type is printed, e.g. `ValueErrror` vs `OSError`.

With f-strings, you do that like this: `f"{my_val!r}"`.

Reply:
Thanks! Still learning Python3